server.py

```diff
@@ -1,24 +1,66 @@
-from flask import Flask, redirect, render_template, url_for
+import click
 import time
-import RPi.GPIO as GPIO
+from flask import Flask, redirect, render_template, url_for, abort
+from users import users
+
+
+class Door(object):
+    def __init__(self, pin):
+        try:
+            import RPi.GPIO as GPIO
+            GPIO.setmode(GPIO.BCM)
+            GPIO.setup(pin, GPIO.OUT)
+            self.gpio = True
+        except ImportError:
+            self.gpio = False
+
+    def open(self, timeout):
+        if self.gpio:
+            GPIO.output(18, True)
+            time.sleep(timeout)
+            GPIO.output(18, False)
+        else:
+            print "opening door for %ds" % timeout
+            time.sleep(timeout)
 
-GPIO.setmode(GPIO.BCM)
-GPIO.setup(18, GPIO.OUT)
 
 app = Flask(__name__)
+door = Door(18)
 
 
 @app.route('/')
-def index():
-    return render_template('main.html')
+@app.route('/<token>')
+def index(token=''):
+    if token not in users:
+        abort(401)
+    return render_template('main.html', token=token, username=users[token])
+
+
+@app.route("/press/")
+@app.route("/press/<token>")
+def press(token=''):
+    if token not in users:
+        abort(401)
+    door.open(3)
+    return redirect(url_for('index', token=token))
```

Comment:
do you have a dead simple (also plain js) way to replace sleep and redirect with something more responsive using XMLHttpRequest?